### DISCUSSION TITLE
How to architect database as OOP string

### DISCUSSION THREAD

**Discussion by jojo8356** [agronholm/sqlacodegen#311] [Q&A]
Hello GitHub community,

I need help getting a database architecture as an OOP string using SQLAlchemy in Python. I tried to do it myself, but I couldn't.

The idea is to convert the database structure into Python OOP classes, for example:

```python
class Account(base_account):
    __tablename__ = 'account'
    id = Column(Integer, primary_key=True)
    name = Column(Text)
    password = Column(Text)```

**Comment by agronholm** [comment]
What's an OOP string?

This project can create declarative classes out of a database structure. Is that not what you're after?

**Reply by jojo8356** [reply]
I'm trying to find out what the option is in the terminal to convert the db into a string like the bit of code I've shown you

**Reply by agronholm** [reply]
I'm not understanding you. Have you followed the instructions in the documentation and installed sqlacodegen? Have you run it? What is not working?

**Reply by jojo8356** [reply]
It's okay, I succeeded, it's just my code that I did wrong
